studio/src/aem/aem-tag-picker-field.js

```diff
@@ -0,0 +1,293 @@
+        const root = new Map();
+
+        tags.forEach((tag) => {
+            const path = tag.path.replace(this.#tagsRoot, '');
+            const parts = path.split('/');
+            let currentLevel = root;
+
+            parts.forEach((part, index) => {
+                if (!currentLevel.has(part)) {
+                    currentLevel.set(part, {
+                        __info__: index === parts.length - 1 ? tag : null,
+                        __children__: new Map(),
+                    });
+                }
+                currentLevel = currentLevel.get(part).__children__;
+            });
+        });
+
+        return root;
+    }
+
+    async toggleTag(path) {
+        await this.#data;
+        const index = this.value.indexOf(path);
+
+        if (!this.multiple) {
+            // If multiple is false, replace the entire value
+            this.value = [path];
+            return;
+        }
+
+        // For multiple selection, toggle the clicked path
+        const currentValue = [...(this.value || [])];
+
+        if (index === -1) {
+            currentValue.push(path);
+        } else {
+            currentValue.splice(index, 1);
+        }
+
+        this.value = currentValue;
+    }
+
+    async #handleChange(event) {
+        const path = event.target.value;
+        event.target.value = '';
+        this.selected = path;
+        this.toggleTag(path);
+    }
+
+    #deleteTag(event) {
+        const pathToDelete = event.target.dataset.path;
+        this.toggleTag(pathToDelete);
+    }
+
+    #resolveTagTitle(path) {
+        const tag = this.#data.get(path);
+        return tag ? tag.title : '';
+    }
+
+    renderSidenavItems(node, parentPath = '') {
+        return [...node.entries()].map(([key, item]) => {
+            const hasChildren = item.__children__.size > 0;
+            const info = item.__info__;
+            const label = info ? this.#resolveTagTitle(info.path) : key;
+            const value = info ? info.path : `${parentPath}/${key}`;
+            return html`
+                <sp-sidenav-item label="${label}" value="${value}">
+                    ${hasChildren
+                        ? this.renderSidenavItems(item.__children__, value)
+                        : html``}
+                    ${hasChildren
+                        ? html`<sp-icon-labels slot="icon"></sp-icon-labels>`
+                        : html`<sp-icon-label slot="icon"></sp-icon-label>`}
+                </sp-sidenav-item>
+            `;
+        });
+    }
+
+    get tagsInHierarchy() {
+        return this.value.filter((path) => path.startsWith(this.#tagsRoot));
+    }
+
+    get tags() {
+        if (this.ready === false) return nothing;
+        if (this.tagsInHierarchy.length === 0) return nothing;
+        return html` ${repeat(
+            this.tagsInHierarchy,
+            (path) => path,
+            (path) =>
+                html`<sp-tag
+                    deletable
+                    @delete=${this.#deleteTag}
```

Comment:
do you think we can add `protected-namespaces` property that contains all subname spaces for which we should _not_ be able to remove tags from UI?

Reply:
for a tag picker, we can restrict the exact tree, the user can add / remove tags from.
And here, only those tags will be shown.
To me it makes sense that the user can remove tags that he might have added.
We can have multiple tag pickers storing tags in the same `tags` field.